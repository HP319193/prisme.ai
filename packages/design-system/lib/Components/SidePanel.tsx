import { Layout } from '../index';
import React from 'react';

export interface SidePanelProps {
  children: React.ReactNode;
  className?: string;
  Header?: React.ReactNode;
}

const SidePanel = ({ Header, className, children }: SidePanelProps) => (
  <Layout Header={Header} className={`p-2 h-full ${className}`}>
    <div className="flex border border-gray-200 border-solid grow h-full rounded p-4">
      {children}
    </div>
  </Layout>
);

export default SidePanel;
