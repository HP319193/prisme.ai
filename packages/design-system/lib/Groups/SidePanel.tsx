import React, { ReactElement } from 'react';
import { Layout, MenuTab } from '../';

export interface SidePanelProps {
  children: ReactElement;
}

const SidePanel = ({ children }: SidePanelProps) => (
  <Layout
    Header={<MenuTab items={['Apps', 'Automations']} onSelect={() => {}} />}
    className="m-2"
  >
    <div className="border border-gray-200 border-solid grow h-full rounded p-4">
      {children}
    </div>
  </Layout>
);

export default SidePanel;
