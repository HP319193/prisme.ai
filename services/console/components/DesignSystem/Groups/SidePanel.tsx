import React from "react";
import { Layout, MenuTab, Button, SidePanelAutomations } from "../index";
import { Typography } from "antd";

export const Title = Typography.Title;

const AutomationsHeader = (
  <div className="flex justify-between items-center">
    <Title level={4} className="mb-0">
      Automations
    </Title>
    <Button>+ Add an automation</Button>
  </div>
);
const AutomationsContent = (
  <Layout
    className="border border-gray border-solid grow h-full rounded p-4 m-2"
    Header={AutomationsHeader}
  >
    {SidePanelAutomations}
  </Layout>
);

export interface SidePanelProps {}

const SidePanel = ({}: SidePanelProps) => (
  <Layout
    Header={<MenuTab items={["Apps", "Automations"]} onSelect={() => {}} />}
  >
    {AutomationsContent}
  </Layout>
);

export default SidePanel;
