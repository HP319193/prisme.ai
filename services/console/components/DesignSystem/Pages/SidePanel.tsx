import React from "react";
import { Layout, MenuTab, Title, Button } from "../";
import SidePanelAutomation from "./SidePanelAutomation";

const AutomationsHeader = (
  <div className="flex justify-between items-center">
    <Title level={4} className="mb-0">
      Automations
    </Title>
    <Button>+ Add an automation</Button>
  </div>
);
const AutomationsContent = (
  <Layout
    className="border border-gray border-solid grow h-full rounded p-4 m-2"
    Header={AutomationsHeader}
  >
    {SidePanelAutomation}
  </Layout>
);

const SidePanel = () => (
  <Layout
    Header={<MenuTab items={["Apps", "Automations"]} onSelect={() => {}} />}
  >
    {AutomationsContent}
  </Layout>
);

export default SidePanel;
