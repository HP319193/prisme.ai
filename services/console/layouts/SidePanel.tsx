import { FC } from 'react';
import ClickAwayListener from 'react-click-away-listener';

const noop = () => {};
interface SidePanelProps {
  sidebarOpen: boolean;
  onClose?: () => void;
}

export const SidePanel: FC<SidePanelProps> = ({
  sidebarOpen,
  onClose = noop,
  children,
}) => {
  return (
    <ClickAwayListener onClickAway={onClose}>
      <div
        className={`
          absolute
          surface-section
          top-0
          left-100
          bottom-0
          p-4
          transition-transform
          transition-duration-100
          transition-ease-in
          ${sidebarOpen ? '-translate-x-100' : ''}
          shadow-4
          z-4`}
      >
        {children}
      </div>
    </ClickAwayListener>
  );
};

export default SidePanel;
