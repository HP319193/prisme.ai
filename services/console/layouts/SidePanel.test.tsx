import SidePanel from "./SidePanel";
import renderer, { act } from "react-test-renderer";
import ClickAwayListener from "react-click-away-listener";

it("should render", () => {
  const root = renderer.create(<SidePanel sidebarOpen={false} />);
  expect(root.toJSON()).toMatchSnapshot();
});

it("should render opened", () => {
  const root = renderer.create(<SidePanel sidebarOpen />);
  expect(root.toJSON()).toMatchSnapshot();
});

it("should close", () => {
  const onClose = jest.fn();
  const root = renderer.create(<SidePanel sidebarOpen onClose={onClose} />);
  act(() => {
    root.root.findByType(ClickAwayListener).props.onClickAway();
  });
  expect(onClose).toHaveBeenCalled();
});

it("should close without crashing", () => {
  const root = renderer.create(<SidePanel sidebarOpen />);
  act(() => {
    root.root.findByType(ClickAwayListener).props.onClickAway();
  });
});
