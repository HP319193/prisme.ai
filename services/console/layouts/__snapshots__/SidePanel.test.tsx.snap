// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="
          absolute
          surface-section
          top-0
          left-100
          bottom-0
          p-4
          transition-transform
          transition-duration-100
          transition-ease-in
          
          shadow-4
          z-2"
  onClick={[Function]}
  onTouchEnd={[Function]}
/>
`;

exports[`should render opened 1`] = `
<div
  className="
          absolute
          surface-section
          top-0
          left-100
          bottom-0
          p-4
          transition-transform
          transition-duration-100
          transition-ease-in
          -translate-x-100
          shadow-4
          z-2"
  onClick={[Function]}
  onTouchEnd={[Function]}
/>
`;
